src/catchup/CatchupManagerImpl.cpp

```diff
@@ -153,14 +154,14 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
             return;
         }
 
-        addToSyncingLedgers(ledgerData);
+        addAndTrimSyncingLedgers(ledgerData);
         logAndUpdateCatchupStatus(true);
         return;
     }
 
     // Next, we buffer every out of sync ledger to allow us to get back in sync
     // in case the ledgers we're missing are received.
-    addToSyncingLedgers(ledgerData);
+    addAndTrimSyncingLedgers(ledgerData);
```

Comment:
I think you introduced a regression here:
if `ledgerData` is a checkpoint ledger `N`, trim will be called, which will see that as the latest checkpoint ledger, which will cause the code below to see as "lowest checkpoint ledger" `N`, and because it waits for `N+1` to actually trigger catchup, it will wait.

So I think the "Trim" code should be updated so that it only deletes the previous ledgers if we have enough to catchup (so there is a ledger after the "highest checkpoint" ledger)